execute/exectypes/costly_messages.go

```diff
@@ -364,4 +376,55 @@ func (c *CCIPMessageExecCostUSD18Calculator) computeExecutionCostUSD18(
 	return cost
 }
 
+// computeDataAvailabilityCostUSD18 computes the data availability cost of a message in USD18s.
+func computeDataAvailabilityCostUSD18(
+	dataAvailabilityFee *big.Int,
+	daConfig cciptypes.DataAvailabilityGasConfig,
+	message cciptypes.Message,
+) plugintypes.USD18 {
+	if dataAvailabilityFee == nil || dataAvailabilityFee.Cmp(big.NewInt(0)) == 0 {
+		return big.NewInt(0)
+	}
+
+	messageGas := calculateMessageMaxDAGas(message, daConfig)
+	return big.NewInt(0).Mul(messageGas, dataAvailabilityFee)
+}
+
+// calculateMessageMaxDAGas calculates the total DA gas needed for a CCIP message
+func calculateMessageMaxDAGas(
+	msg cciptypes.Message,
+	daConfig cciptypes.DataAvailabilityGasConfig,
+) *big.Int {
+	// Calculate token data length
+	var totalTokenDataLen int
+	for _, tokenAmount := range msg.TokenAmounts {
+		totalTokenDataLen += len(tokenAmount.SourcePoolAddress) +
+			len(tokenAmount.DestTokenAddress) +
+			len(tokenAmount.ExtraData) +
+			EvmWordBytes +
+			len(tokenAmount.DestExecData)
+	}
+
+	// Calculate total message data length
+	dataLen := ConstantMessagePartBytes +
+		len(msg.Data) +
+		len(msg.Sender) +
+		len(msg.Receiver) +
+		len(msg.ExtraArgs) +
+		len(msg.FeeToken) +
+		EvmWordBytes*2 + // FeeTokenAmount and FeeValueJuels
```

Comment:
ExtraArgs /FeeToken/FeeTokenAmount/FeeValueJuels are not posted to dest chain, so we don't need to account for them here;

I think `Receiver` is already accounted for in the overhead too, so you just need
```
overhead + len(data) + len(sender) + len_token
```